frontend/app/src/next/hooks/use-run-detail.tsx

```diff
@@ -85,10 +87,16 @@ function RunDetailProviderContent({
   const [lastRefetchTime, setLastRefetchTime] = useState(Date.now());
   const { toast } = useToast();
 
+  // Memoize query keys to prevent unnecessary refetches
+  const runDetailsQueryKey = useMemo(
+    () => ['workflow-run-details:get', runId],
+    [runId],
+  );
```

Comment:
this seems like premature optimization IMO - it's a single value?

Reply:
ya good catch, was debugging something else and this snuck in